connectors/sources/google_cloud_storage.py

```diff
@@ -378,11 +384,19 @@ async def get_docs(self, filtering=None):
         Yields:
             dictionary: Documents from Google Cloud Storage.
         """
-        async for buckets in self.fetch_buckets():
-            if not buckets.get("items"):
-                continue
-            async for blobs in self.fetch_blobs(
-                buckets=buckets,
-            ):
-                for blob_document in self.get_blob_document(blobs=blobs):
-                    yield blob_document, partial(self.get_content, blob_document)
+        if self.configuration["buckets"] == ["*"]:
+            async for buckets in self.fetch_buckets():
+                if not buckets.get("items"):
+                    continue
+                async for blobs in self.fetch_blobs(
+                    buckets=buckets,
+                ):
+                    for blob_document in self.get_blob_document(blobs=blobs):
+                        yield blob_document, partial(self.get_content, blob_document)
+        else:
+            for bucket in self.configuration["buckets"]:
+                async for blobs in self.fetch_blobs(
+                    buckets={"items": [{"id": bucket, "name": bucket}]},
+                ):
+                    for blob_document in self.get_blob_document(blobs=blobs):
+                        yield blob_document, partial(self.get_content, blob_document)
```

Comment:
One more suggestion, you can make a change to `fetch_buckets()`:

```python
async def fetch_buckets(self, buckets):
    if "*" in buckets:
        async for bucket in self._google_storage_client.api_call(
            resource="buckets",
            method="list",
            full_response=True,
            project=self._google_storage_client.user_project_id,
            userProject=self._google_storage_client.user_project_id,
        ):
            if bucket.get("items")
                yield bucket
    else:
        for bucket in buckets:
            yield {"items": [{"id": bucket, "name": bucket}]}
```

Then you have simplify this method:
```suggestion
        async for bucket in self.fetch_buckets(self.configuration["buckets"]):
            async for blobs in self.fetch_blobs(buckets=bucket):
                for blob_document in self.get_blob_document(blobs=blobs):
                    yield blob_document, partial(self.get_content, blob_document)
```